be/src/storage/tablet_reader.cpp

```diff
@@ -57,14 +65,17 @@ void TabletReader::close() {
 
 Status TabletReader::prepare() {
     SCOPED_RAW_TIMER(&_stats.get_rowsets_ns);
-    std::shared_lock l(_tablet->get_header_lock());
-    auto st = _tablet->capture_consistent_rowsets(_version, &_rowsets);
-    if (!st.ok()) {
-        _rowsets.clear();
-        std::stringstream ss;
-        ss << "fail to init reader. tablet=" << _tablet->full_name() << "res=" << st;
-        LOG(WARNING) << ss.str();
-        return Status::InternalError(ss.str().c_str());
+    Status st = Status::OK();
+    if (_rowsets.empty()) {
+        std::shared_lock l(_tablet->get_header_lock());
```

Comment:
!_rowsets.empty()  typo?  is it tested?

Reply:
No, non-empty `_rowsets` indicates that it is captured before creating this TabletReader.
Therefore, only empty `_rowsets` need to capture rowsets in this if block.

Reply:
I'd like to add some comments.

Reply:
so it is a second-time chance rowsets capturing?  add comment to explain why, it looks so weird.